src/CoreBundle/Bridge/Chameleon/Migration/Script/update-1553067741.inc.php

```diff
@@ -0,0 +1,28 @@
+<h1>Build #1553067741</h1>
+<h2>Date: 2019-03-20</h2>
+<div class="changelog">
+    - Remove Flash support
+</div>
+<?php
+
+$databaseConnection = TCMSLogChange::getDatabaseConnection();
+
+$flashUrlHandlerId = $databaseConnection->fetchColumn("SELECT `id` FROM `cms_smart_url_handler` WHERE `name` = 'TCMSSmartURLHandler_FlashCrossDomain'");
+if (false === $flashUrlHandlerId) {
+    return;
+}
+
+$data = TCMSLogChange::createMigrationQueryData('cms_smart_url_handler_cms_portal_mlt', 'en')
+  ->setWhereEquals([
+      'source_id' => $flashUrlHandlerId,
+  ])
+;
+TCMSLogChange::delete(__LINE__, $data);
+
+$data = TCMSLogChange::createMigrationQueryData('cms_smart_url_handler', 'en')
+  ->setWhereEquals([
+      'id' => $flashUrlHandlerId,
+  ])
+;
+TCMSLogChange::delete(__LINE__, $data);
+
```

Comment:
I think we need to handle the now deprecated SWF files in the media library.
I would not delete them, but list them to the admin, so they may be archived if needed and then deleted manually.

Reply:
I'm not sure what happens when the system tries to display yet-existing Flash files in the backend. And they could still be linked in text fields. So yes, we should check this.

Reply:
Rechecked what happens after migration. In the media manager the files can still be viewed, just without the preview image. Flash files used in text fields are still displayed in the frontend, as the object tag is rendered on save in the table editor. So nothing explodes and the default behavior is OK I think.

Therefore I just added a note in the upgrade guide, so that developers/administrators remove usages manually (in the rare case that there are massive amounts of Flash usages developers need to be creative).